### PR TITLE
Add bc assignment to object list

### PR DESCRIPTION
**Pull request by lorenzovecchietti**

### CODE REVIEW

File: pyaedt/icepak.py [hunk 1]
```diff
@@ -3672,8 +3672,9 @@ def assign_source(
 
         Parameters
         ----------
-        assignment : int or str
-            If int, Face ID. If str, object name.
+        assignment : int or str or list
+            If int, Face ID. If str, object name. It is also possible to provide a list of Face IDs
+            and object names.
```
Comment (PipKat) [inline pyaedt/icepak.py:3677]:
```suggestion
            If an integer is provided, it must be a face ID. If a string is provided, it must
            be an object name. It is also possible to provide a list of face IDs
            and object names.
```
Can the list contain both face IDs and object names? If not, change to ",,, a list of face IDs or object names."

File: pyaedt/icepak.py [hunk 2]
```diff
@@ -3895,8 +3898,8 @@ def assign_solid_block(
 
         Parameters
         ----------
-        object_name : str
-            Name of the object.
+        object_name : str or list
+            Name of the object or list of names.
```
Comment (PipKat) [inline pyaedt/icepak.py:3902]:
```suggestion
            Object name or a list of object names.
```

File: pyaedt/icepak.py [hunk 3]
```diff
@@ -4022,8 +4027,8 @@ def assign_hollow_block(
...
```
Comment (PipKat) [inline pyaedt/icepak.py:4031]:
```suggestion
            Object name or a list of object names.
```

File: pyaedt/icepak.py [hunk 4]
```diff
@@ -4254,7 +4262,8 @@ def assign_free_opening(
         Parameters
         ----------
         assignment : int or str
```
Comment (PipKat) [inline pyaedt/icepak.py:4264]:
```suggestion
        assignment : int or str or list
```

File: pyaedt/icepak.py [hunk 5]
```diff
@@ -4420,7 +4429,8 @@ def assign_pressure_free_opening(
...
```
Comment (PipKat) [inline pyaedt/icepak.py:4431]:
```suggestion
        assignment : int or str or list
```

File: pyaedt/icepak.py [hunk 6]
```diff
@@ -4420,7 +4429,8 @@ def assign_pressure_free_opening(
...
-            Integer indicating face ID or object name string.
+            Integer indicating face ID or object name string. Also list of Face IDs or object names
+            is accepted.
```
Comment (PipKat) [inline pyaedt/icepak.py:4433]:
```suggestion
           Integer indicating a face ID or a string indicating an object name. A list of face
           IDs or object names is also accepted.
```

File: pyaedt/icepak.py [hunk 7]
```diff
@@ -4488,7 +4498,8 @@ def assign_velocity_free_opening(
         Parameters
         ----------
         assignment : int or str
```
Comment (PipKat) [inline pyaedt/icepak.py:4500]:
```suggestion
        assignment : int or str or list
```

File: pyaedt/icepak.py [hunk 8]
```diff
@@ -4563,7 +4574,8 @@ def assign_mass_flow_free_opening(
...
```
Comment (PipKat) [inline pyaedt/icepak.py:4576]:
```suggestion
        assignment : int or str or list
```

File: pyaedt/icepak.py [hunk 9]
```diff
@@ -4563,7 +4574,8 @@ def assign_mass_flow_free_opening(
...
-            Integer indicating face ID or object name string.
+            Integer indicating face ID or object name string. Also list of Face IDs or object names
+            is accepted.
```
Comment (PipKat) [inline pyaedt/icepak.py:4578]:
```suggestion
           Integer indicating a face ID or a string indicating an object name. A list of face
           IDs or object names is also accepted.
```

File: pyaedt/icepak.py [hunk 10]
```diff
@@ -4563,7 +4574,8 @@ def assign_mass_flow_free_opening(
...
         boundary_name : str, optional
             Boundary name. Default is ``None``, in which case the name is generated automatically.
```
Comment (PipKat) [inline pyaedt/icepak.py:4580]:
```suggestion
            Boundary name. The default is ``None``, in which case the name is generated automatically.
```

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**